IncludeToolbox/IncludeWhatYouUse/IWYU.cs

```diff
@@ -263,64 +256,69 @@ static public string RunIncludeWhatYouUse(string fullFileName, EnvDTE.Project pr
+                                clangOptions += "-march=arm ";
+                                break;
+                            case VCProjectUtils.Base.TargetMachineType.MIPS:
+                                clangOptions += "-march=mips ";
+                                break;
+                            case VCProjectUtils.Base.TargetMachineType.THUMB:
+                                clangOptions += "-march=thumb ";
+                                break;*/
                     }
                 }
 
-                // icwyu options
+
+                iwyuOptions += "-Xiwyu --verbose=" + settings.LogVerbosity + " ";
+                if (settings.MappingFiles.Length == 0)
                 {
-                    process.StartInfo.Arguments += "-Xiwyu --verbose=" + settings.LogVerbosity + " ";
-                    for (int i = 0; i < settings.MappingFiles.Length; ++i)
-                        process.StartInfo.Arguments += "-Xiwyu --mapping_file=\"" + settings.MappingFiles[i] + "\" ";
-                    if (settings.NoDefaultMappings)
-                        process.StartInfo.Arguments += "-Xiwyu --no_default_mappings ";
-                    if (settings.PCHInCode)
-                        process.StartInfo.Arguments += "-Xiwyu --pch_in_code ";
-                    switch (settings.PrefixHeaderIncludes)
-                    {
-                        case IncludeWhatYouUseOptionsPage.PrefixHeaderMode.Add:
-                            process.StartInfo.Arguments += "-Xiwyu --prefix_header_includes=add ";
-                            break;
-                        case IncludeWhatYouUseOptionsPage.PrefixHeaderMode.Remove:
-                            process.StartInfo.Arguments += "-Xiwyu --prefix_header_includes=remove ";
-                            break;
-                        case IncludeWhatYouUseOptionsPage.PrefixHeaderMode.Keep:
-                            process.StartInfo.Arguments += "-Xiwyu --prefix_header_includes=keep ";
-                            break;
-                    }
-                    if (settings.TransitiveIncludesOnly)
-                        process.StartInfo.Arguments += "-Xiwyu --transitive_includes_only ";
-
-                    // Set max line length so something large so we don't loose comment information.
-                    // Documentation:
-                    // --max_line_length: maximum line length for includes. Note that this only affects comments and alignment thereof,
-                    // the maximum line length can still be exceeded with long file names(default: 80).
-                    process.StartInfo.Arguments += "-Xiwyu --max_line_length=1024 ";
-
-                    // Custom stuff.
-                    process.StartInfo.Arguments += settings.AdditionalParameters;
-                    process.StartInfo.Arguments += " ";
+                    settings.MappingFiles = new string[] { "stl.c.headers.imp", "msvc.imp", "boost-all.imp", "boost-all-private.imp" };
```

Comment:
Why are you manipulating the settings outside of the settings screen?
Also, much more important, where do you get this list from? Not having those files will result in errors while running the tool. On top of that for me only absolute paths work which is why the automatic download puts the filenames as such into the settings.

Reply:
Good spot. There is no need. I will fix that.